### PR TITLE
Visualization Module I: Density Estimation and Classification

### PR DESCRIPTION
**Pull request by Vince505**
@krsgpp 

First release of the visualization Module. Only density estimation is implemented.
Waiting for revision

### CODE REVIEW

File: datadriven/src/sgpp/datadriven/datamining/base/SparseGridMinerCrossValidation.cpp [hunk 1]
```diff
@@ -95,7 +97,7 @@ double SparseGridMinerCrossValidation::learn(bool verbose) {
         if (verbose) {
           std::ostringstream out;
           out << "###############"
-              << "Itertation #" << (iteration++) << std::endl
+              << "Itertation #" << (iteration) << std::endl
```
Comment (krsgpp) [inline datadriven/src/sgpp/datadriven/datamining/base/SparseGridMinerCrossValidation.cpp]: Let's fix the "Itertation" here too :-)

**Comment by valentjn** [comment]
There are lots of C-style allocations in here (`malloc`, `calloc`), which should be replaced by C++ `new`s and `delete`s. In addition, `unique_ptr` should be used to reduce the risk of memory leaks.

**Comment by valentjn** [comment]
Are these `system` calls really necessary to make or delete a directory? Especially in combination with user input one has to be very careful; AFAIK `system` uses a shell to execute the commands, which means that by calling `Visualizer::createFolder` with `foo; rm -rf /` as argument (don't try it) I could wipe out my disk. Since SG++ is used as a library, this would be a security risk if maliciously exploited.

Can you use the respective syscalls instead, e.g., `mkdir` from `sys/stat.h` on POSIX? Same with removing files.

File: datadriven/src/sgpp/datadriven/datamining/builder/MinerFactory.cpp [hunk 2]
```diff
@@ -24,19 +24,22 @@
 #include <sgpp/datadriven/scalapack/BlacsProcessGrid.hpp>
 
 #include <string>
+#include <iostream>
```
Comment (valentjn) [inline datadriven/src/sgpp/datadriven/datamining/builder/MinerFactory.cpp]: Remove leftover `#include`.
```suggestion
```

File: datadriven/src/sgpp/datadriven/datamining/configuration/DataMiningConfigParser.cpp [hunk 3]
```diff
@@ -34,6 +34,7 @@
 #include <sgpp/datadriven/datamining/modules/fitting/FitterTypeParser.hpp>
 #include <sgpp/datadriven/datamining/modules/scoring/ScorerMetricTypeParser.hpp>
 #include <sgpp/solver/TypesSolver.hpp>
+#include <sgpp/datadriven/datamining/modules/visualization/VisualizationTypesParser.hpp>
 
 #include <map>
```
Comment (valentjn) [inline datadriven/src/sgpp/datadriven/datamining/configuration/DataMiningConfigParser.cpp]: Add missing `#include`.
```suggestion
#include <iostream>
#include <map>
```

File: datadriven/src/sgpp/datadriven/datamining/modules/visualization/VisualizationParameters.hpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * CrossvalidationConfiguration.hpp
+ *
+ *  Created on: Jun 8, 2019
+ *      Author: Vincent Bautista
+ */
+
+#pragma once
+
+#include <iostream>
```
Comment (valentjn) [inline datadriven/src/sgpp/datadriven/datamining/modules/visualization/VisualizationParameters.hpp]:
```suggestion
```

File: datadriven/src/sgpp/datadriven/datamining/modules/visualization/VizualizerDummy.cpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * VisualizerDensityEstimation.hpp
+ *
+ *  Created on: 16th Jun 2019
+ *      Author: Vincent Bautista
+ */
+
+#include <sgpp/datadriven/datamining/modules/visualization/VisualizerDummy.hpp>
+#include <sgpp/datadriven/datamining/modules/fitting/ModelFittingBase.hpp>
+#include <iostream>
```
Comment (valentjn) [inline datadriven/src/sgpp/datadriven/datamining/modules/visualization/VizualizerDummy.cpp]:
```suggestion
```

File: datadriven/src/sgpp/datadriven/datamining/configuration/DataMiningConfigParser.hpp [hunk 6]
```diff
@@ -39,6 +42,8 @@ using sgpp::base::AdaptivityConfiguration;
 using sgpp::base::GeneralGridConfiguration;
 using sgpp::datadriven::CrossvalidationConfiguration;
 using sgpp::datadriven::DensityEstimationConfiguration;
+using sgpp::datadriven::VisualizationGeneralConfig;
+using sgpp::datadriven::VisualizationParameters;
 using sgpp::solver::SLESolverConfiguration;
```
Comment (valentjn) [inline datadriven/src/sgpp/datadriven/datamining/configuration/DataMiningConfigParser.hpp]: Avoid `using` in headers.

File: datadriven/src/sgpp/datadriven/tools/CSVTools.hpp [hunk 7]
```diff
@@ -18,6 +18,9 @@
 #include <string>
 #include <vector>
 
+using sgpp::base::DataMatrix;
+using sgpp::base::DataVector;
```
Comment (valentjn) [inline datadriven/src/sgpp/datadriven/tools/CSVTools.hpp]: Avoid `using` in headers.

**Pull request merged**